lib/ts/styles/styleContext.tsx

```diff
@@ -33,35 +34,40 @@ const StyleContext = React.createContext<NormalisedStyle>({
 export function StyleProvider({
     children,
     styleFromInit,
+    rootStyleFromInit,
     getDefaultStyles,
     defaultPalette,
     rawPalette,
 }: {
     children: JSX.Element;
     styleFromInit?: Styles;
+    rootStyleFromInit: Styles;
     getDefaultStyles: (palette: NormalisedPalette) => NormalisedDefaultStyles;
     defaultPalette: NormalisedPalette;
     rawPalette: Record<string, string>;
 }): JSX.Element {
     const palette = getMergedPalette(defaultPalette, rawPalette);
 
-    const styles: NormalisedStyle = {
-        palette,
-        ...getDefaultStyles(palette),
-    };
+    let mergedStyles = getDefaultStyles(palette);
+    if (rootStyleFromInit !== undefined) {
+        // Palette is a reserved word, delete it if exists.
+        delete rootStyleFromInit.palette;
+
+        mergedStyles = getMergedStyles(mergedStyles, rootStyleFromInit);
+    }
+
     if (styleFromInit !== undefined) {
         // Palette is a reserved word, delete it if exists.
         delete styleFromInit.palette;
 
-        Object.keys(styleFromInit).forEach((key) => [
-            (styles[key] = {
-                ...styles[key],
-                ...styleFromInit[key],
-            }),
-        ]);
```

Comment:
Why is this removed?

Reply:
It's done by `mergedStyles = getMergedStyles(mergedStyles, styleFromInit);`

Reply:
Which upon reviewing it again a bit misleading: it actually writes into the first parameter instead of creating a new object (and it's used everywhere like it creates a new object).

Reply:
That's odd.. not sure why this would have been done in the first place if `getMergedStyles` existed before. hmm..

Reply:
Well, it was a tiny bit different because the `style` object it was merging into had an extra palette property, but I don't think that would've been an issue either. The only difference I can find is that this uses `Object.keys` and other one uses a `for..in` loop which also enumerates inherited properties.
Can I resolve this?